crates/sui-core/src/consensus_types/consensus_output_api.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::consensus_types::AuthorityIndex;
+
+pub(crate) trait ConsensusOutputAPI {
+    fn reputation_score_sorted_desc(&self) -> Option<Vec<(AuthorityIndex, u64)>>;
```

Comment:
Can the return type just be `Vec<u64>` or `Option<Vec<u64>>`, where empty means unavailable and index corresponds to `AuthorityIndex`?

Reply:
We want both index and score though.

Reply:
Just for discussion, I think the caller can sort the scores themselves, so this can just be a vec of scores.

Reply:
Indeed we could. Let me know what you think but I chose to use it because there are a bunch of other callers to `ReputationScore::authorities_by_score_desc()`, so we don't want to get rid of this function anyway, it's a waste to not use it.